File: BOJ(Java 공부용)/11561.java
import java.io.*;
import static java.lang.Long.max;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().split(" ")[0]);
        for(int tc = 1; tc<=t ;tc++) {
            long n = Long.parseLong(br.readLine().split(" ")[0]);
            n *= 2;
            long left = 1, right = 1000000000, ans = 0;
            while(left<=right) {
                long mid = (left + right) / 2;
                long temp = (mid+1)*mid;
                if(temp > n ) right = mid-1;
                else {
                    ans = max(ans,mid);
                    left = mid + 1;
                }
            }
            bw.write(ans+"\n");
        }
        bw.flush();
        bw.close();
    }
}

File: BOJ 2/11561.cpp
#include <cstdio>
#include <algorithm>
#include <cstdint>
#include <vector>
typedef int32_t Int;
typedef int64_t ll;
using namespace std;
int main() {
	Int t;
	scanf("%d", &t);
	while (t--) {
		ll n;
		scanf("%lld", &n);
		n *= 2;
		ll left = 1, right = 1e9,ans = 0;
		while (left <= right) {
			ll mid = (left + right) / 2;
			ll temp = (mid + 1) * mid;
			if (temp > n) right = mid - 1;
			else {
				ans = max(ans,mid);
				left = mid + 1;
			}
		}
		printf("%lld\n", ans);
	}
	return 0;
}